server/models/annotation.py

```diff
@@ -158,94 +143,84 @@ def filter(self, annotation, user=None, additionalKeys=None):
-                featuresetGlobalFeatures = {
-                    feature['id']: feature
-                    for feature in
-                    featureset['globalFeatures']
-                }
-                featuresetLocalFeatures = {
-                    feature['id']: feature
-                    for feature in
-                    featureset['localFeatures']
-                }
-                if featuresetLocalFeatures:
-                    image = Image().load(doc['meta']['imageId'], force=True)
-                    superpixels = Image().superpixelsData(image)
-                    maxSuperpixel = superpixels.max()
-                else:
-                    maxSuperpixel = None
-                for featureId, featureValue in six.viewitems(
-                        doc['meta']['annotations']):
-                    if featureId in featuresetGlobalFeatures:
-                        featureOptions = set(
-                            option['id']
-                            for option in
-                            featuresetGlobalFeatures[featureId]['options'])
-                        if featureValue not in featureOptions:
-                            raise ValidationException(
-                                'Annotation feature "%s" has invalid '
-                                'value "%s".' % (featureId, featureValue))
-                    elif featureId in featuresetLocalFeatures:
-                        if not (
-                            isinstance(featureValue, list) and
-                            len(featureValue) == maxSuperpixel + 1 and
-                            all(superpixelValue in [0.0, 0.5, 1.0]
-                                for superpixelValue in featureValue)
-                        ):
-                            raise ValidationException(
-                                'Annotation feature "%s" has invalid '
-                                'value "%s".' % (featureId, featureValue))
-                    else:
+                            'Annotation feature "%s" has invalid value "%s".' %
+                            (featureId, featureValue))
+                elif featureId in featuresetLocalFeatures:
+                    if not (
+                        isinstance(featureValue, list) and
+                        len(featureValue) == maxSuperpixel + 1 and
+                        all(superpixelValue in [0.0, 0.5, 1.0]
+                            for superpixelValue in featureValue)
+                    ):
                         raise ValidationException(
-                            'Annotation has invalid feature "%s".' % featureId)
+                            'Annotation feature "%s" has invalid value "%s".' %
+                            (featureId, featureValue))
+                else:
+                    raise ValidationException(
+                        'Annotation has invalid feature "%s".' % featureId)
 
         return super(Annotation, self).validate(doc)
```

Comment:
This should probably be removed since `Model.validate()` raises a `NotImplementedError`.